### ISSUE TITLE
Missing referenced type named "C" in section 2.2, "POD Data Types"

### ISSUE THREAD

**Issue by tahonermann** [itanium-cxx-abi/cxx-abi#190]
[Section 2.2 "POD Data Types"](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#pod) states:
> A member pointer type is treated exactly as if it were the C type [described below](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#member-pointers).

The "described below" link references [section 2.3, "Member Pointers"](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#member-pointers). That section does not contain any prose but it does have two subsections, [section 2.3.1, "Data Member Pointers"](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#data-member-pointers) and [section 2.3.2, Member Function Pointers"](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#member-function-pointers), neither of which defines a type named "C".

[Section 2.4, "Non-POD Class Types"](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#class-types) uses the name "C" as a generic identifier for a class type. If this is the type intended by the "C type" reference, then the "described below" link should be updated to reference this section. I don't see an obvious way in which the "C" type described there fits with the "member pointer type is treated exactly as if it were the C type" wording though, so it seems that some other elucidation is still required.

**Comment by jensmaurer** [comment]
My guess is that "C type" refers to this C-style type definition presented further below:

```
struct {
    fnptr_t ptr;
    ptrdiff_t adj;
  };
```

**Comment by tahonermann** [comment]
Thanks, Jens. That seems likely. There would still be a gap for pointers to data members though.

**Comment by jensmaurer** [comment]
A data member is explained to be an offset stored in a `ptrdiff_t`.

**Comment by tahonermann** [comment]
Indeed, but the "A member pointer type is treated exactly as if it were the C type described below" wording doesn't differentiate between data members and member functions. It seems that "the C type described below" must be intended to refer to different parts of the prose; for pointers to data members, the C type is `ptrdiff_t` where as for pointers to member functions, the C type is the unnamed struct type below.

In 2.3.1 Data Member Pointers:
>  The basic ABI properties of data member pointer types are those of `ptrdiff_t`. 

In 2.3.2 Member Function Pointers:
>  In all representations, the basic ABI properties of member function pointer types are those of the following class, where `fnptr_t` is the appropriate function-pointer type for a member function of this type:
> 
>   ```
>   struct {
>     fnptr_t ptr;
>     ptrdiff_t adj;
>   };
>  ```

This would make sense as the intent. I think the wording can still be improved though.

**Comment by rjmccall** [comment]
That is the intent, yes. I'm open to suggestions about how to improve the wording.

**Comment by zygoloid** [comment]
Maybe just remove the "C":

> A member pointer type is treated exactly as if it were the <del>C</del> type [described below](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#member-pointers).

**Comment by tahonermann** [comment]
> Maybe just remove the "C":

That helps, but there are two types to choose from (for data members vs member functions). Perhaps:

> A member pointer type is treated exactly as if it were <u><b>one of</b></u> the <s>C</s> type<u><b>s</b></u> [described below](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#member-pointers).
